lib/code_fund_web/controllers/dashboard_controller.ex

```diff
@@ -10,14 +10,20 @@ defmodule CodeFundWeb.DashboardController do
     secret_key = Application.get_env(:code_fund, CodeFundWeb.Endpoint)[:metabase_secret_key]
 
     payload =
-      case Enum.member?(current_user.roles, "admin") do
-        true ->
+      cond do
```

Comment:
Instead of a cond and hardcoding ids here, can we do something like: 
```
config :metabase,
  dashboard_mappings: [admin: 1, sponsor: 3, user: 2]
  
  
defmodule Metabase.Helpers do
  defp dashboard_mapping(role), do: Application.get_env(:metabase, :dashboard_mappings) |> Keyword.get(role)
    
  def get_metabase_details(%User{roles: ["admin" | _roles]}),  do: %{resource: %{dashboard: dashboard_mapping(:admin)}, params: %{}}
  def get_metabase_details(%User{id: id, roles: [_roles | "sponsor"]}),  do: %{resource: %{dashboard: dashboard_mapping(:sponsor)}, params: %{user_id: id}}
  def get_metabase_details(%User{id: id, roles: _roles}),  do: %{resource: %{dashboard: dashboard_mapping(:user)}, params: %{user_id: id}}
end

```
In your controller code
current_user
|> Map.put(:roles, Enum.sort(current_user.roles))
|> Metabase.Helpers.get_metabase_details


```
Or alternatively, the Enum.member code is almost what we have already in require_any_role.ex#user_has_role? so I'm wondering if it's possible to move this into a public scope, test it and reuse it.

Reply:
I think I'll leave this refactor for you. I agree it makes sense, but there's too much to do before we get into refactors like this (for me)